### ISSUE TITLE
Place Submit button to finish form on all tabs

### ISSUE THREAD

**Issue by lexislav** [verbb/formie#353]
**What are you trying to do?**

Creating a form surveys, I have run into a scenario when I would like to offer user to finish the form on any tab.
Not only on the last one.

Is there a recommended way to do this? Like custom template to render buttons?

Thanks for any idea for the solution.

**Comment by engram-design** [comment]
It's slightly clunky at the moment, rather than just a button click, but if you pass in a post param with `completeSubmission` is will force the submission to be completed and done.

You should be able to create a button like the below to action this.

```
<button name="completeSubmission" value="true">Finish Form</button>
```

**Comment by lexislav** [comment]
Method mentioned above worked until latest update to formie 1.3.26

<button class="fui-btn fui-submit" name="completeSubmission" value="true"><i class="fas fa-paper-plane mr-i"></i>{{ "Submit"|t }}</button>

So this just go to next form page instead of finalize the form and reload the page.
Any idea what has been chenge and if there is a way to fix this?

Thanks

**Comment by engram-design** [comment]
I believe this is an issue when you have client-side validation enabled, because it hijacks the submission process to validate the form first, then submit. However, because it's a programatic submit, it loses the context of which button was triggered. As browsers only send through button values for the button that was clicked (to handle multiple buttons), and because this is a programatic submit, after client-side validation, it doesn't know which button was pressed.

However, I've added handling for this for the next release. To get the fix early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-3 as 1.3.26",
  "...": "..."
}
```

Then run `composer update`.

**Comment by lexislav** [comment]
I have tried update to suggested dev version of formie,
the form has Validate form on Submit and Validate When Typing off.

But the manually implemented Submit/Finish the form still goes to the next form page.

Other form settings are:
- Page Reload (Server Side)
- Loading indicator - none

Spam protection: on
- reCaptcha on, show on all pages
(But have tried to turn it off with no effect).

Hopefully this helps.

**Comment by engram-design** [comment]
Do note that you'll need to specify the type to be `submit`:

`<button name="completeSubmission" type="submit" value="true">Finish Form</button>`

**Comment by lexislav** [comment]
I have updated the button.
The form seems to be send now, but the form is not redirected to specified URL.

Form setting is:
- Page reload (server side)
- redirect to url: formpage?form=thanks

But the page is reloaded without specified parameter.
So it is getting close…

My current html arroudn buttons - there are some custom classes:

**Comment by engram-design** [comment]
@lexislav Just pushed another fix for this in https://github.com/verbb/formie/commit/2a76a268f5362d5ffaccaacba394a1d7266c8596

**Comment by lexislav** [comment]
It works! Thanks

**Comment by engram-design** [comment]
Fixed in [1.3.27](https://github.com/verbb/formie/releases/tag/1.3.27)
